### PR TITLE
Add component Button

### LINKED ISSUE #7: Component Button
## Description
Clickable button containing an icon and text, normal and small size

## Design
[Link](https://www.figma.com/file/L4dBE9w4Hc6a6e6PLQiNC9/Perfect-Pic?node-id=2%3A3)

## Acceptance criteria

- [ ] props: icon = '', text = '', small=false, noBorder=false, onClick
- [ ] Show button with optional icon and/or text, color: light-dimmed, background: primary-gradient
- [ ] Standard version: font-size: inputFontSize, iconHeight=1.5rem, padding: 0.75rem, gap 0.5rem
- [ ] Small version: font-size: 1rem, iconHeight=1rem, padding: 0.5rem, gap 0.5rem
- [ ] noBorder: Icon/Text only without a background, e.g. for the trash CTA

## Blocked by
#2 
#4 

## Size
S

### PR DESCRIPTION
**Pull request by dirkzukunft**
Resolves #7

https://perfectpic-7-button-chczmj6pxn.herokuapp.com/storybook/?path=/story/components-button--button

Note: Since I merged the needed Icon component branch here, its commits are shown here, too.

### CODE REVIEW

File: src/app/components/Button/Button.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import { ICONS } from '../Icon/icons';
+import type { ButtonProps } from './Button';
+import Button from './Button';
+
+export default {
+  title: 'Components/Button',
+  component: Button,
+  argTypes: {
+    icon: {
+      options: [``, ...Object.keys(ICONS)],
```
Comment (phlgr) [inline src/app/components/Button/Button.stories.tsx:13]:
```suggestion
      options: Object.keys(ICONS),
```
Wouldn't this be sufficient?

File: src/app/components/Button/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  onClick,
+  className = '',
+}: ButtonProps): JSX.Element {
+  const { colorClass } = COLORS[color];
+
+  return (
+    <button
+      onClick={onClick}
+      className={`${styles.button} ${colorClass} 
+        ${small ? styles.small : ``} ${transparent ? styles.transparent : ``} ${className}`}
```
Comment (phlgr) [inline src/app/components/Button/Button.tsx:33]:
```suggestion
        ${small && styles.small} ${transparent && styles.transparent} ${className}`}
```

**Review by phlgr — CHANGES_REQUESTED** [review]

File: src/app/components/Button/Button.stories.tsx (hunk 1 above)
Comment (dirkzukunft) [reply src/app/components/Button/Button.stories.tsx:13]: "No icon" is a relevant option here, too

Comment (dirkzukunft) [reply src/app/components/Button/Button.stories.tsx:13]: I changed the Icon component and added a "none" key

**Pull request merged**